<launch>
    <arg name="output" default="screen"/>

    <arg name="scout_count" default="2" />
    <arg name="excavator_count" default="2" />
    <arg name="hauler_count" default="2" />

    <node name="main_controller" pkg="nasa_srcp_final" type="main_controller.py" output="$(arg output)" >
        <param name="team_id" value="1" />
    </node>

    <group ns="/small_scout_1">
        <node name="base_controller" pkg="nasa_srcp_final" type="robot_controller.py" output="$(arg output)" >
            <param name="robot_name" value="/small_scout_1" />
        </node>
    </group>

    <group ns="/small_excavator_1">
        <node name="base_controller" pkg="nasa_srcp_final" type="robot_controller.py" output="$(arg output)" >
            <param name="robot_name" value="/small_excavator_1" />
        </node>
    </group>

    <group ns="/small_hauler_1">
        <node name="base_controller" pkg="nasa_srcp_final" type="robot_controller.py" output="$(arg output)" >
            <param name="robot_name" value="/small_hauler_1" />
        </node>
    </group>

</launch>